src/resolve/resolved.c

```diff
@@ -67,13 +67,18 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
-        /* Drop privileges, but keep three caps. Note that we drop those too, later on (see below) */
-        r = drop_privileges(uid, gid,
-                            (UINT64_C(1) << CAP_NET_RAW)|          /* needed for SO_BINDTODEVICE */
-                            (UINT64_C(1) << CAP_NET_BIND_SERVICE)| /* needed to bind on port 53 */
-                            (UINT64_C(1) << CAP_SETPCAP)           /* needed in order to drop the caps later */);
-        if (r < 0)
-                goto finish;
+        /* Drop privileges, but only if we have been started as root. If we are not running as root we assume all
+         * privileges are already dropped. */
+        if (getuid() == 0) {
```

Comment:
good question, and i don't have a clear answer for this. Ultimately we know exactly that we will be called with uid==euid, as it is systemd that spawns resolved, hence the distinction doesn't matter too much. I decided to use getuid() here simply because it is the more obvious one, and if euid/uid really don't match, then maybe it could even be assumed that something is set up in some way and we shouldn't alter it...